sway/config/input.c

```diff
@@ -156,7 +158,7 @@ struct input_config *store_input_config(struct input_config *ic) {
 		merge_input_config(current, ic);
 		free_input_config(ic);
 		ic = current;
-	} else if (!wildcard) {
+	} else if (!wildcard && !type_wildcard) {
 		sway_log(SWAY_DEBUG, "Adding non-wildcard input config");
 		i = list_seq_find(config->input_configs, input_identifier_cmp, "*");
 		if (i >= 0) {
```

Comment:
If this is not a type config and a matching type config exists, that should be used as the base here